### PR TITLE
chore: check if object is converted to string and stringify it

### PR DESCRIPTION
**Pull request by BonapartePC**
## What are the changes introduced in this PR?

check if object is converted to string and stringify it

Write a brief explainer on your code changes.

## What is the related Linear task?

Resolves [PIPE-1518](https://linear.app/rudderstack/issue/PIPE-1518/store-context-traits-in-a-single-column-for-malformed-payloads)

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3726/test-report.html

File: src/warehouse/index.js [hunk 1]
```diff
@@ -208,6 +208,20 @@ function setDataFromInputAndComputeColumnTypes(
   level = 0,
 ) {
   if (!input || !isObject(input)) return;
+  if (completePrefix.endsWith('context_traits_') && isStringLikeObject(input)) {
+    if (prefix === 'context_traits_') {
```
Comment (lvrach) [inline src/warehouse/index.js:212]: Here we only handle `context_traits_`, but I was under the impression we need to handle `traits_` as well. (for group calls)

Comment (BonapartePC) [reply src/warehouse/index.js]: Added support for traits in group calls

File: test/__tests__/warehouse.test.js [hunk 2]
```diff
@@ -1233,4 +1233,119 @@ describe("isBlank", () => {
       expect(isBlank(testCase.input)).toEqual(testCase.expected);
     });
   }
-});
\ No newline at end of file
+});
+
+describe("context/group traits", () => {
+  const testCases = [
```
Comment (lvrach) [inline test/__tests__/warehouse.test.js:1239]: [minor] It might be a bit more readable if we separate the tests for group events.

**Review by lvrach — APPROVED** [review]
Please add linear ticket in the description

File: src/warehouse/index.js [hunk 3]
```diff
@@ -208,6 +208,23 @@ function setDataFromInputAndComputeColumnTypes(
...
   if (!input || !isObject(input)) return;
+  if (
+    (completePrefix.endsWith('context_traits_') || completePrefix === 'group_traits_') &&
+    isStringLikeObject(input)
+  ) {
+    if (prefix === 'context_traits_') {
+      appendColumnNameAndType(
+        utils,
+        eventType,
+        `${prefix.length > 0 ? prefix : completePrefix}`,
```
Comment (achettyiitr) [inline src/warehouse/index.js:219]: We are already checking for prefix === 'context_traits_'. `prefix.length` will always be greater than 0.

**Review by achettyiitr — APPROVED** [review]

File: test/__tests__/warehouse.test.js (hunk 2 above)
Comment (BonapartePC) [reply test/__tests__/warehouse.test.js]: Done

File: src/warehouse/index.js (hunk 3 above)
Comment (BonapartePC) [reply src/warehouse/index.js]: Done

**Review by achettyiitr — APPROVED** [review]

**Review by Gauravudia — APPROVED** [review]

**Pull request merged**